File: src/game/entities/entity.ts
/*
  Author: Alias
  Date: 29/3/2021

  Base class for static map objects.
*/
import {Vector2} from "../data_structures/vector2";
import {MapObject} from "../map_object";

export class Entity extends MapObject {
  constructor(x: number, y: number) {
    this.setPositionV2(new Vector2(x, y));
  }
}

File: src/game/traits/entity.rs
mod data_structures;

use self::data_structures::*;

/// Base trait for objects that are on the map
pub trait Entity {
    /// Return the owner, or None if it's not owned
    fn get_owner(&self) -> Option<usize>;
    /// Returns the UID for the object
    fn get_id(&self) -> usize;
    /// Current position of the entity on the map
    fn get_position() -> Vector2;
}